### PR TITLE
Improve performance for `pointless-exception-statement`

### LINKED ISSUE #8073: Performance regression: `pointless-exception-statement`
Thanks for the new check! I've just tested it with Home-Assistant and was able to find a handful of issues. Unfortunately, the current implementation has significant performance problems.

Comparing 56121344f6c9c223d8e4477c0cdb58c46c840b4b with 89a20e2ccdb2927ac370f7c6663fc83318988e10, a full pylint ci run can take between `6%` and `43%` more time. The spread is likely a result of different caches as `-j2` is used for the run.

```
before: 7:59
after: 8:29 - 11:35 (multiple runs)
```

_Originally posted by @cdce8p in https://github.com/PyCQA/pylint/pull/7939#pullrequestreview-1252510286_

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Improve performance for `pointless-exception-statement` especially for large code bases.
Use 3 step approach outline in https://github.com/PyCQA/pylint/pull/7939#discussion_r1072865819 to catch almost all issues without the negative performance impact while still providing an option for users to check more / everything if they use a custom exception name pattern.

1. Check common names. Almost all exceptions contain one the these words: `Exception`, `Error`, or `Warning`.
2. Check not yet covered stdlib exceptions. https://docs.python.org/3/library/exceptions.html#exception-hierarchy
_If a 3rd party library has some other names, those could be added here as well. Although I would limit it to fairly popular ones, something like `requests` for example._
3. Allow users to define additional regex names to check.

Alternative to #8078
Closes #8073

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Looks pretty good, what do you think @jayaddison ?  (We need to add you as co-author if we merge this imo)

File: pylint/checkers/base/basic_checker.py [hunk 1]
```diff
@@ -265,6 +276,18 @@ class BasicChecker(_BasicChecker):
...
     }
+    options = (
+        (
+            "additional-exception-classes",
+            {
+                "default": [],
+                "type": "regexp_csv",
+                "metavar": "<pattern>[,<pattern>...]",
+                "help": "List of regular expressions of additional "
+                "exception names to check for 'pointless-exception-statement",
```
Comment (cdce8p) [inline pylint/checkers/base/basic_checker.py:287]: I'm open to suggestions if someone has a better name for it

**Comment by jayaddison** [comment]
@cdce8p why is the default to perform non-thorough exception matching, with thorough matching only enabled by adding `.*` or similar as a pattern string?

I think that smaller codebases are likely to be the more common case (and it's more important for them to build good foundations as they grow), so we should provide for that - but allowing large codebases to reduce the thoroughness of checks (if they choose to) for performance reasons seems fine, too.

**Comment by cdce8p** [comment]
Closing this one as #8078 is the better approach.
For details see: https://github.com/PyCQA/pylint/issues/8073#issuecomment-1397664994

**Pull request closed without merging**